### ISSUE TITLE
gtests/test_nopython_callprofiler.cpp timing is unstable

### ISSUE THREAD

**Issue by yungyuc** [solvcon/solvcon#347]
See https://github.com/solvcon/modmesh/actions/runs/9378360062/job/25821452032#step:9:275 :

```
/Users/runner/work/modmesh/modmesh/gtests/test_nopython_callprofiler.cpp:175: Failure
Value of: diff_time(node1->data().total_time, uniqueTime1 + uniqueTime2)
  Actual: false
Expected: true
```

The test should use a more stable way to test for time.

**Comment by tigercosmos** [comment]
@yungyuc  I checked all scheduled actions, no errors before, it's probably the PR issue.
https://github.com/solvcon/modmesh/actions?page=3&query=event%3Aschedule

**Comment by yungyuc** [comment]
Right, it is a random failure.  Please take a look at my quote and link in the issue description.

**Comment by tigercosmos** [comment]
@yungyuc  No, it's not a random failure, it never happens in the past few months

**Comment by yungyuc** [comment]
I see what you meant.  It does not happen with scheduled runs, but only happens with PR runs?

**Comment by tigercosmos** [comment]
@yungyuc  yes, that's what I meant. The current code should be very stable in the master branch, so the PR owner need to check why it failed.

**Comment by yungyuc** [comment]
This is good finding.  Thanks, @tigercosmos .  Let me unassign this issue from you and move it out in progress.  We should revisit when we find another occasion.

**Comment by tigercosmos** [comment]
We don't have this issue now.
